cmd/build/v2/build.go

```diff
@@ -253,6 +260,39 @@ func (bc *OktetoBuilder) Build(ctx context.Context, options *types.BuildOptions)
 	return options.Manifest.ExpandEnvVars()
 }
 
+func (ob *OktetoBuilder) calculateBuildContextHash(buildInfo *model.BuildInfo) string {
+	buildContext := buildInfo.Context
+	treeHash, err := ob.repo.GetTreeHash(buildContext)
```

Comment:
```suggestion
	treeHash, err := ob.Config.GetTreeHash(buildContext)
```
and calculate it after the repoCommit ?

Reply:
build context hash cannot be calculated when repo commit is calculated because we need to know the service to generate this data, that is not the case to get git repo commit hash